cli/azd/cmd/cobra_builder.go

```diff
@@ -58,28 +59,58 @@ func (cb *CobraBuilder) BuildCommand(descriptor *actions.ActionDescriptor) (*cob
 		}
 	}
 
-	// Configure action resolver for leaf commands
-	if !cmd.HasSubCommands() {
-		if err := cb.configureActionResolver(cmd, descriptor); err != nil {
-			return nil, err
-		}
+	// Configure action resolver for all commands
+	if err := cb.configureActionResolver(cmd, descriptor); err != nil {
```

Comment:
I wonder if we adopt @weikanglim's proposal for solving this at the cobra layer, we might be able to remove stuff like this?

Reply:
I need to talk to @weikanglim and see if there's something I might be missing.  AFAK, using `PersistentPreRunE` only work when there's is a `RunE` or `Run` definition (all of the PersistentPre/ Pre / Post are hooks to the `RunE`. When there's no RunE, those hooks won't trigger).

For example, we have `PersistentPreRunE` on root command, and it is only ever triggered when a runnable command is executed (for example, `azd version`).  But if we run a non-runnable command, like `azd pipeline`, the `PersistentPreRunE` is not executed, as there's no `RunE` for `azd pipeline`.

That's basically the reason why I am adding `RunE` to all commands, because for our use-case.

However, I will explore using the HelpFn a little more.  If we can tell cobra to see `--docs` as a help flag, just like `-h` or `--help`, then we can do everything within the HelpFn

Reply:
Cobra does not allow to set one extra parameter as sentinel for help.  Cobra uses `help` and `-h` in a hardcoded way.
It only allows setting the help template and a function to actually write the help. But all of this is trigerer only by either `--help` or `-h`